gnomad_qc/v4/create_release/calculate_variant_statistics.py

```diff
@@ -24,188 +51,218 @@
+        missense, and synonymous.
+    :param rare_variants: Include count of rare variants, defined as those which have
+        adj AF <0.1%.
+    :param vep_canonical: If `by_csqs` is True, filter to only canonical transcripts.
+        If trying count variants in all transcripts, set it to False. Default is True.
+    :param vep_mane: If `by_csqs` is True, filter to only MANE transcripts. Default is
+        False.
+    :return: Table containing per-sample variant counts.
     """
-    vds = get_gnomad_v4_vds(test=test, release_only=True, split=True)
+    logger.info("Filtering input MT to variants in the supplied annotation HT...")
+    mt = mt.semi_join_rows(annotation_ht)
 
-    # Read in release data and select relevant annotations.
-    ht = release_sites(data_type=data_type).ht()
-    ht = ht.select("freq", "filters", "region_flags", "vep")
-
-    if test:
-        logger.info("Test: filtering to variants on chr22")
-        ht = ht.filter(ht.locus.contig == "chr22")
-
-    logger.info("Filter VDS to variants in gnomAD release.")
-    vds = hl.vds.filter_variants(vds, ht.select())
-
-    # Create variant matrix table from VDS and annotate with all relevant info.
-    vmt = vds.variant_data
-    logger.info("Variant MT created and checkpointing, without annotations")
-    vmt = vmt.checkpoint(
-        new_temp_file(f'vmt_{"TEST" if test else ""}', extension="mt"),
-        overwrite=overwrite,
+    # Add extra Allele Count and Allele Type annotations to variant MatrixTable,
+    # according to Hail standards, to help their computation.
+    mt = mt.annotate_rows(
+        **vmt_sample_qc_variant_annotations(global_gt=mt.GT, alleles=mt.alleles)
     )
 
-    # Add extra Allele Count and Allele Type annotations to variant Matrix
-    # Table, according to Hail standards, to help their computation.
-    ac_and_atype = vmt_sample_qc_variant_annotations(
-        global_gt=vmt.GT, alleles=vmt.alleles
-    )
-    vmt = vmt.annotate_rows(**ac_and_atype)
-
+    keep_annotations = ["freq", "filters", "region_flags"]
     if by_csqs:
-        ht = filter_vep_transcript_csqs(
-            ht,
+        annotation_ht = filter_vep_transcript_csqs(
+            annotation_ht,
             synonymous=False,
             canonical=vep_canonical,
             mane_select=vep_mane,
         )
+        annotation_ht = get_most_severe_consequence_for_summary(annotation_ht)
+        keep_annotations.extend(["most_severe_csq", "lof"])
 
-        ht = get_most_severe_consequence_for_summary(ht)
-
-        ht = ht.select("freq", "filters", "region_flags", "most_severe_csq", "lof")
-
-    # NOTE: not size efficient, but we are never checkpointing THIS I don't believe
-    vmt = vmt.annotate_rows(**ht[vmt.row_key])
+    # Annotate the MT with the needed annotations.
+    annotation_ht = annotation_ht.select(*keep_annotations).checkpoint(
+        hl.utils.new_temp_file("annotation_ht", "ht")
+    )
+    mt = mt.annotate_rows(**annotation_ht[mt.row_key])
```

Comment:
I really really don't want to checkpoint the MT if we don't need to, checkpointing the full MT is a lot. The test is slow because I didn't also filter the HT to only the test samples, that is an easy fix.

Reply:
But the annotation_ht has no samples inside, you mean test variants?

Reply:
yeah, I meant variants, I was thinking about something else when I wrote samples

Reply:
added the filter to chr22